### ISSUE TITLE
changed binding to STB_GLOBAL

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#1190]
https://reviews.llvm.org/D90108 broke LLVM_IAS=1 for x86_64, arm64, and arm.  Filing an issue to track the fixes, since we'll need to track they all get fixed and backported to 4.19.

cc @MaskRay @jcai19 @samitolvanen 

x86 patch is accepted: https://git.kernel.org/pub/scm/linux/kernel/git/tip/tip.git/commit/?id=4d6ffa27b8e5116c0abb318790fd01d4e12d75e6
arm64 patch is in mainline: ec9d78070de986ecf581ea204fd322af4d2477ec
perf patch submitted: https://lore.kernel.org/lkml/20201104005609.1316230-1-maskray@google.com/

I think 32b arm is still a problem:

```
$ ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- make LLVM=1 LLVM_IAS=1 -j71 
  AS      arch/arm/lib/memcpy.o
  AS      arch/arm/lib/memmove.o
  AS      arch/arm/lib/memset.o
<unknown>:0: error: memcpy changed binding to STB_GLOBAL
make[1]: *** [scripts/Makefile.build:364: arch/arm/lib/memcpy.o] Error 1
make[1]: *** Waiting for unfinished jobs....
<unknown>:0: error: memmove changed binding to STB_GLOBAL
<unknown>:0: error: memset changed binding to STB_GLOBAL
```

**Comment by MaskRay** [comment]
I'll start fixing arm now...

**Comment by nickdesaulniers** [comment]
The 32b ARM failure is currently linux-next specific, due to this commit: https://lore.kernel.org/linux-arm-kernel/20201019084140.4532-3-linus.walleij@linaro.org/

**Comment by MaskRay** [comment]
I don't know how to reference a commit in linux-next but not in mainline...
Just replied to the thread by Linus Walleij (I could reply because @nathanchance kindly CCed it to clang-built-linux...)

**Comment by nathanchance** [comment]
> I don't know how to reference a commit in linux-next but not in mainline...

Same way:

Fixes: <sha> ("<commit title>")

If the maintainer's tree is not stable (i.e. they rebase), they will usually squash the fix in or they will need to adjust the SHA for the fixes commit. If it is the former, the tag won't matter regardless and if it is the latter, it is their burden to fix up the SHA in the Fixes tag, not yours (because there is nothing you can do).

The vast majority of maintainers will not rebase so their SHAs are constant.

**Comment by MaskRay** [comment]
Thanks:)

Sent [[PATCH] ARM: Change arch/arm/lib/mem*.S to use WEAK instead of .weak](http://lore.kernel.org/r/20201105181542.854788-1-maskray@google.com)

**Comment by nathanchance** [comment]
x86_64 patch has been merged into mainline: https://git.kernel.org/linus/4d6ffa27b8e5116c0abb318790fd01d4e12d75e6

**Comment by nickdesaulniers** [comment]
perf patch: https://git.kernel.org/pub/scm/linux/kernel/git/acme/linux.git/commit/?h=tmp.perf/urgent&id=db1a8b97a0a36155171dbb805fbcb276e07559f6

**Comment by nickdesaulniers** [comment]
backports sent for x86_64 on 5.4 and 4.19: https://lore.kernel.org/stable/CAKwvOdkK1LgLC4ChptzUTC45WvE9-Sn0OqtgF7-odNSw8xLTYA@mail.gmail.com/T/#u

**Comment by lazerl0rd** [comment]
> backports sent for x86_64 on 5.4 and 4.19: https://lore.kernel.org/stable/CAKwvOdkK1LgLC4ChptzUTC45WvE9-Sn0OqtgF7-odNSw8xLTYA@mail.gmail.com/T/#u

I'll drop my temporary Linux 4.19 patch and merge this to avoid confusion.

**Comment by nickdesaulniers** [comment]
arm64 patch ec9d78070de986ecf581ea204fd322af4d2477ec landed in v5.10-rc2.
63632bd5dd31485450834948155be98703aaccb7 in v5.9.5.
ca16a42f5f0da332a1756e7b48b228b6d3bcf24b in v5.4.76.
DNE in v4.19.y.

conflicts: 3ac0f4526dfb80625f5c2365bccd85be68db93ef v5.6-rc1
35e61c77ef386555f3df1bc2057098c6997ca10b v5.6-rc1

backport sent: https://groups.google.com/g/clang-built-linux/c/jHD0eQXFPGk

**Comment by nickdesaulniers** [comment]
arm: 735e8d93dc2b107f7891a9c2b1c4cfbea1fcbbbc (v5.11-rc1)
arm64: ec9d78070de986ecf581ea204fd322af4d2477ec (v5.10-rc2)
x86_64: 4d6ffa27b8e5116c0abb318790fd01d4e12d75e6 (v5.10-rc3)
perf: db1a8b97a0a36155171dbb805fbcb276e07559f6 (v5.10-rc5)
